framework/CodeInterfaces/PHISICS/ControlParser.py

```diff
@@ -0,0 +1,151 @@
+"""
+Created on July 11th, 2017
+@author: rouxpn
+"""
+
+import os
+import sys
+import re 
+from shutil import copyfile 
+import fileinput 
+from decimal import Decimal 
+import time
+import xml.etree.ElementTree as ET
+from random import *
+
+class ControlParser():
+
+  """
+    this opens the various INSTANT and MRTAU inputs to orientate the interface. 
+    It provides the optional flag activated in the input and verifies the various inputs agree with each other
+  """
+  def replaceValues(self, genericXMLdict):
+    """
+    replace the values from the pertured dict and put them in the deconstructed original dictionary
+    """
+    setXML = set(genericXMLdict)
+    #print setXML
+    setPertDict = set(self.pertDict)
+    #print setPertDict
+    for key in setPertDict.intersection(setXML):
+      genericXMLdict[key] = self.pertDict.get(key, {})
+    #print genericXMLdict
+    return genericXMLdict
+
+  def dictFormating_from_perturbed_to_generic(self, XMLdict):
+    """
+    Transform the ditionary comning from the XML input into the templated dictionary.
+    The templated format is {DENSITY|FUEL|ISOTOPE}
+    """
+    genericXMLdict = {}
+    #print XMLdict
+    for paramXML in XMLdict.iterkeys():
+      for matXML in XMLdict.get(paramXML).iterkeys():
+        for isotopeXML, densityValue in XMLdict.get(paramXML).get(matXML).iteritems():
+          genericXMLdict[paramXML.upper()+'|'+matXML.upper()+'|'+isotopeXML.upper()] = densityValue 
+    #print genericXMLdict
+    return genericXMLdict
+
+  def dictFormating_from_XML_to_perturbed(self):
+    """
+    Transform the dictionary of dictionaries from the XML tree to a dictionary of dictionaries
+    formatted identically as the perturbed dictionary 
+    the perturbed dictionary template is {'DENSITY':{'FUEL':{'ISOTOPE'}}}
+    """
+    # declare the dictionaries 
+    XMLdict = {}
+    matList = []
+    isotopeList = []
+    XMLdict['density'] = {}
+    for matXML in self.root.getiterator('mat'):
+      #print matXML.attrib.get('id')
+      for isotopeXML in self.root.getiterator('isotope'):
+        matList.append(matXML.attrib.get('id'))
+        isotopeList.append(isotopeXML.attrib.get('id'))
+    #print matList
+    #print isotopeList
+    for i in xrange(0,len(matList)):
```

Comment:
xrange is not easily portable to Python 3...please, use ```range```. In addition, the iteration always starts from ```0```. there is no need to explicitly set the ```start``` to 0